contrib/internet-urllib3.py

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env python
+
+import urllib3
+import certifi
+
+def isInternetConnected(url="www.ietf.org"):
+  result = False
+  http = urllib3.PoolManager(
+    cert_reqs='CERT_REQUIRED', # Force certificate check.
+    ca_certs=certifi.where(),  # Path to the Certifi bundle.
+  )
+  try:
+    r = http.request('HEAD', 'https://' + url)
+    result = True
+  except Exception as e:  # urllib3.exceptions.SSLError
+    result = False
```

Comment:
The only things that can go wrong are: 
1. Cannot find `cacert.pem`.  Solution, show a warning to the user, at the start of the app. This is god, because no sync is interrupted.  Normally this will never happen, because every operating system is supposed to ship with a `cacert.pem` this is what enables trustworthy secure internet communications. Probably only when a virus maybe deletes your `cacert.pem` will you see this `Error, cannot start sync, cannot find your system's root certificate store file`.
2. Outdated `cacert.pem`. This happens from time to time, every few months, the package containing the root cert store file needs to be updated, with `apt`, `yum`, etc.  It's a good thing to stop the sync for this reason, high security is very important.
3. The app tries to verify connection to internet and receives a fake cert in the reply. Again, this is another case where you really should stop the sync before it even starts, and notify the user, the server they're trying to connect to is returning a bad cert, and is possibly insecure, and the cert should be fixed.  I believe the app should protect the user credentials and not transmit his IMAP password and username credentials to a remote IMAP server which is trying to use a fake SSL cert!